src/api/records/record.controller.js

```diff
@@ -855,10 +855,13 @@ export async function uniqueValuesCollection(req, res) {
 
 export async function validate(req, res) {
   let query = {};
-  if (req.params.taxID !== 'all') query = { taxID: +req.params.taxID };
-  const records = await Record.find(query)
-    // .limit(200000)
-    .cursor();
+
+  if (req.params.taxID !== 'all') {
+    const taxID = req.params.taxID.split(',');
+    query = { taxID };
```

Comment:
Did you test it? I believe this is missing the [$in](https://docs.mongodb.com/v3.6/reference/operator/query/in/) operator

Reply:
Yes, i tested and it works. Probably it is not necessary to use the `$in` operator considering that by passing an array of values to taxID attribute, mongoose make this conversion into a raw query.

I used these test cases:

- http://localhost:7070/v2/records/validate/6909,7238,7204
- http://localhost:7070/v2/records/validate/7204
- http://localhost:7070/v2/records/validate/all

it would be of great if you can test and let me know your opinion. It is possible i am missing something o making some mistake.

Reply:
Oh ok, It looks rare to me, but you're right about mongoose [converting the query](https://mongoosejs.com/docs/tutorials/query_casting.html#implicit-in), I don't see anything missing, but I'll test it and let you know.